Xamarin.Forms.Core/BindingBase.cs

```diff
@@ -129,17 +123,39 @@ internal virtual object GetSourceValue(object value, Type targetPropertyType)
 			if (value == null && TargetNullValue != null)
 				return TargetNullValue;
 
-			if (StringFormat != null)
-				return string.Format(StringFormat, value);
+			if (StringFormat != null && TryFormat(StringFormat, value, out var formatted))
```

Comment:
we don't want binding code to throw in this case. we want logging, and ideally BindingDiagnostic